### ISSUE TITLE
ContentFields missing if part is welded on via ActivatingFilter

### ISSUE THREAD

**Issue by Xceno** [OrchardCMS/Orchard#5197]
Hey guys!
I was wondering: I have a part with two fields attached to it. 
Right now I attach this part to my ContentType inside the handler via ActivatingFilter<T>.
My problem is: the fields only show up when i attach the part via migrations and I just can't find out what exactly is different.
Has anyone some experience with this problem?

PS: My issue seems to be related to this StackOverflow question (which sadly has no answers):
http://stackoverflow.com/questions/23829385/welding-a-contentpart-having-a-contentfield

**Comment by jtkech** [comment]
I agree, in its Weld() method, the ContentItemBuilder try to retrieve the part definition through the content type definition on which we want to add the part. So, a new part is created but with an empty field collection...

For more details and a quick workaround for testing, see the above StackOverflow question

Best

**Comment by Xceno** [comment]
I'm glad you had the time to investigate this further!
I tried your workaround in a quick test and it seems to solve the issue. Nonetheless it feels like a bug that should be adressed.

**Comment by DavidDeVries** [comment]
Thank you, I ran into the same problem and this workaround works. But it should be fixed in the source.
